rickshaw-index

```diff
@@ -823,13 +823,17 @@ if (exists $result{'iterations'}) {
                                                         my $this_end;
                                                         # index_metric() to return the easliest-begin and latest-end for metric types matching the primary-metric
                                                         (my $num_metric_docs_submitted, $this_begin, $this_end) = index_metrics('queue', $metric_file_prefix, $cs_name, $cs_id, $base_metric_doc_ref, $primary_metric);
-                                                        #$num_docs_submitted{'metric_desc'} += $num_metric_docs_submitted;
                                                         # From processing all metric files, get the very-earliest-begin and very-latest-end
-                                                        if (not defined $earliest_begin or $earliest_begin > $this_begin) {
-                                                            $earliest_begin = $this_begin;
-                                                        }
-                                                        if (not defined $latest_end or $latest_end < $this_end) {
-                                                            $latest_end = $this_end;
+                                                        if (defined $this_begin and defined $this_end) {
+                                                            if (not defined $earliest_begin or $earliest_begin > $this_begin) {
+                                                                $earliest_begin = $this_begin;
+                                                            }
+                                                            if (not defined $latest_end or $latest_end < $this_end) {
+                                                                $latest_end = $this_end;
+                                                            }
+                                                        } else {
+                                                            printf "ERROR: index_metrics() returned undefined this_begin and/or undefined this_end, exiting\n";
```

Comment:
What are the conditions under which this would happen?

Reply:
When the wrong primary-metric is used, which still needs to be fixed.  We'll probably keep this just as a sanity check.